syntax = "proto3";

package eu.surething_project.core;

option java_multiple_files = true;
option java_package = "eu.surething_project.core.grpc";
option java_outer_classname = "LocClaimProto";

import "google/protobuf/any.proto";
import "loc_time.proto";
import "signature.proto";

// definition of the location claim
message LocationClaim {
    // unique identifier for claim
    string claimId = 1;
	// prover unique id - can also be a pseudonym
	string proverId = 2;
	// prover's location - in one of alternative location schemas
	Location location = 3;
    // time of location claim -- in one of alternative time schemas
    Time time = 4;
    // type of the evidence that supports the claim - fully-qualified protobuf message name
    string evidenceType = 5;
	// location evidence e.g. sensor readings, photo, screen signing - according to evidence type
	google.protobuf.Any evidence = 6;
}

// signed location claim 
message SignedLocationClaim {
    LocationClaim claim = 1; // claim
	Signature proverSignature = 2; // prover's signature
}

